spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/SetStatusProperties.java

```diff
@@ -0,0 +1,30 @@
+package org.springframework.cloud.gateway.config;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@ConfigurationProperties("spring.cloud.gateway.set-status")
+public class SetStatusProperties {
+
+	private static final String DEFAULT_HEADER_NAME = "original-http-status";
+	private String originalStatusHeaderName;
```

Comment:
Just set the default here. Boot will overwrite if supplied by the user.

Please had one line javadoc comment, this is the documentation for the property that ends up in the appendix.